controllers/managedocs_controller.go

```diff
@@ -977,16 +1062,45 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 			Key: alertRelabelConfigSecretKey,
 		}
 
-		// prometheus tls -- will be uncommented
-		// tlsConfig := r.prometheus.Spec.Web.TLSConfig
-		// tlsConfig.KeySecret.LocalObjectReference.Name = prometheusServiceSecretName
-		// tlsConfig.Cert.Secret.LocalObjectReference.Name = prometheusServiceSecretName
-		// tlsConfig.ClientCA.ConfigMap.LocalObjectReference.Name = servingCertCABundleConfigMapName
+		if deployerCSV, err := r.getCSVByPrefix(deployerCSVPrefix); err == nil {
+			deployerCSVDeployment := &deployerCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0]
+			deployerCSVContainers := deployerCSVDeployment.Spec.Template.Spec.Containers
+			var kubeRbacImage string
+			for key := range deployerCSVContainers {
+				container := deployerCSVContainers[key]
+				if container.Name == "kube-rbac-proxy" {
+					kubeRbacImage = container.Image
+				}
+			}
+
+			prometheusContainers := r.prometheus.Spec.Containers
+			for key := range prometheusContainers {
+				container := &prometheusContainers[key]
+				if container.Name == "kube-rbac-proxy" {
+					container.Image = kubeRbacImage
+				}
+			}
+		} else {
+			return fmt.Errorf("Unable to set image for kube-rbac-proxy container: %v", err)
+		}
```

Comment:
Error handling should come first to remove one label of indentation from the large code block.

```suggestion
		if deployerCSV, err := r.getCSVByPrefix(deployerCSVPrefix); err != nil {
			return fmt.Errorf("Unable to set image for kube-rbac-proxy container: %v", err)
		}
		
		deployerCSVDeployment := &deployerCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0]
		deployerCSVContainers := deployerCSVDeployment.Spec.Template.Spec.Containers
		var kubeRbacImage string
		for key := range deployerCSVContainers {
			container := deployerCSVContainers[key]
			if container.Name == "kube-rbac-proxy" {
				kubeRbacImage = container.Image
			}
		}

		prometheusContainers := r.prometheus.Spec.Containers
		for key := range prometheusContainers {
			container := &prometheusContainers[key]
			if container.Name == "kube-rbac-proxy" {
				container.Image = kubeRbacImage
			}
		}
```